sdk/tools/azure-openrewrite/src/main/java/com/azure/openrewrite/core/http/HttpResponseCustomRecipe.java

```diff
@@ -0,0 +1,110 @@
+package com.azure.openrewrite.core.http;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.NlsRewrite;
+import org.openrewrite.Recipe;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.J;
+
+import com.azure.openrewrite.util.ConfiguredParserJavaTemplateBuilder;
+
+/**
+ * A custom OpenRewrite recipe to migrate the use of HttpResponse.
+ *
+ * <p>This recipe performs the following transformations:</p>
+ * <ul>
+ *   <li>Replaces the getHeaderValue(String) method with getHeaders().getValue(HttpHeaderName.fromString(String)).</li>
+ *   <li>Replaces the getHeaderValue(HttpHeaderName) method with getHeaders().getValue(HttpHeaderName).</li>
+ *   <li>Replaces the getBodyAsBinaryData() method with BinaryData.fromObject(getValue()).</li>
+ * </ul>
+ *
+ * <p>Example transformations:</p>
+ * <pre>
+ * Before: String getHeaderValue(java.lang.String)
+ * After: String getHeaders().getValue(io.clientcore.core.http.models.HttpHeaderName.fromString(java.lang.String))
+ *
+ * Before: java.lang.String getHeaderValue(com.azure.core.http.HttpHeaderName)
+ * After: String getHeaders().getValue(io.clientcore.core.http.models.HttpHeaderName)
+ *
+ * Before: BinaryData getBodyAsBinaryData()
+ * After: BinaryData BinaryData.fromObject(getValue())
+ * </pre>
+ */
+public class HttpResponseCustomRecipe extends Recipe {
+
+    /**
+     * Default constructor for {@link HttpResponseCustomRecipe}.
+     */
+    public HttpResponseCustomRecipe() {
+        super();
+    }
+
+    @Override
+    public @NlsRewrite.DisplayName String getDisplayName() {
+        return "HttpResponseCustomRecipe";
+    }
+
+    @Override
+    public @NlsRewrite.Description String getDescription() {
+        return "The custom recipe for HttpResponse";
+    }
+
+    @Override
+    public JavaIsoVisitor<ExecutionContext> getVisitor() {
+        ConfiguredParserJavaTemplateBuilder templateBuilder = ConfiguredParserJavaTemplateBuilder.defaultBuilder();
+        return new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+                J.MethodInvocation visitedMethod = super.visitMethodInvocation(method, executionContext);
+                MethodMatcher methodMatcher;
+                JavaTemplate replacementTemplate;
+
+                methodMatcher = new MethodMatcher("com.azure.core.http.HttpResponse getHeaderValue(java.lang.String)");
```

Comment:
This is missing the rename from `HttpResponse` to `Response<T>` and also changing the imports. We are missing many other transformations for HttpResponse. It might be good to add a link to the GitHub issue tracking all the transformations still to be done.

Reply:
@srnagar The recipe changing HttpResponse to Response<T> is in yaml. It's in `com.azure.core.http.yml` The java recipes are mainly for what can't be done in yaml. There may still be an issue because of the generic type however that I need to investigate. Tracking here: https://github.com/Azure/azure-sdk-for-java/issues/45483